src/pages/TabDetails.tsx

```diff
@@ -86,89 +101,110 @@ export function useInfoForOneCycle(idx: number): InfoOneCycle {
   const lengthOfPeriod = getAverageLengthOfPeriod(cycles);
 
-  const { t } = useTranslation();
-
   return (
-    <IonItem
-      class="transparent-center"
-      lines="none"
-    >
-      <IonLabel position="stacked">
-        {title ? (
-          <h2>{`${t("Current cycle")}: ${title}`}</h2>
-        ) : (
-          <h2>{t("Current cycle")}</h2>
-        )}
-      </IonLabel>
-      <IonLabel position="stacked">
-        {startDate ? (
-          <p>{`${t("Started date")} ${new Date(
-            startDate,
-          ).toLocaleDateString()}`}</p>
-        ) : (
-          <p>{t("Started date")}</p>
-        )}
-      </IonLabel>
-      <IonLabel position="stacked">
+    <div id="progress-block">
+      <div style={{ marginLeft: "15px" }}>
+        <IonLabel>
+          <p style={lenCycleStyle}>{title}</p>
+        </IonLabel>
         <IonProgressBar
           class="current-progress"
+          style={{ marginTop: "5px", marginBottom: "5px" }}
           value={(lengthOfPeriod / 100) * 3}
           buffer={(progressBarBuffer / 100) * 3}
-        ></IonProgressBar>
-      </IonLabel>
-    </IonItem>
+        />
+        <IonLabel>
+          {t("locale") === "ru" ? (
+            <p style={datesStyle}>{`${format(new Date(startDate), "dd MMMM", {
+              locale: ru,
+            })} - `}</p>
+          ) : (
+            <p style={datesStyle}>{`${format(
+              new Date(startDate),
+              "MMM dd",
+            )} - `}</p>
+          )}
+        </IonLabel>
+      </div>
+    </div>
   );
 };
 
 interface IdxProps {
   idx: number;
 }
 
-const ItemProgress = (props: IdxProps) => {
-  const info = useInfoForOneCycle(props.idx);
+const ListProgress = () => {
+  const cycles = useContext(CyclesContext).cycles;
 
-  return (
-    <IonItem
-      class="transparent-center"
-      lines="none"
-    >
-      <IonLabel position="stacked">
-        <h2>{info.lengthOfCycleString}</h2>
-      </IonLabel>
-      <IonLabel position="stacked">
-        <p>{info.dates}</p>
-      </IonLabel>
-      <IonLabel position="stacked">
-        <IonProgressBar
-          value={(info.lengthOfPeriod / 100) * 3}
-          buffer={(info.lengthOfCycleNumber / 100) * 3}
-        ></IonProgressBar>
-      </IonLabel>
-    </IonItem>
-  );
-};
+  const ItemProgress = (props: IdxProps) => {
+    const info = useInfoForOneCycle(props.idx + 1);
 
-const ListProgress = () => {
-  const numbers = [1, 2, 3, 4, 5];
-  const list = numbers.map((idx) => (
-    <ItemProgress
-      key={idx}
-      idx={idx}
-    />
-  ));
+    return (
+      <div
+        id="progress-block"
+        style={{ marginTop: "15px" }}
+      >
+        <div style={{ marginLeft: "15px" }}>
+          <IonLabel>
+            <p style={lenCycleStyle}>{info.lengthOfCycleString}</p>
+          </IonLabel>
+          <IonProgressBar
+            style={{ marginTop: "5px", marginBottom: "5px" }}
+            value={(info.lengthOfPeriod / 100) * 3}
```

Comment:
You can move this calculation inside `const` above and call it something like `lengthOfPeriodProgress`, `lengthOfCycleProgress`

Reply:
I moved the calculations into separate functions because they are needed in several places. (```setBufferProgress``` and ```setValueProgress``` functions)